src/internal/converters/eml/eml.go

```diff
@@ -183,6 +271,21 @@ func FromJSON(ctx context.Context, body []byte) (string, error) {
 		}
 	}
 
+	// TODO "#microsoft.graph.calendarSharingMessage" (https://github.com/alcionai/corso/issues/5041)
+	otype := ptr.Val(data.GetOdataType())
+	if otype == "#microsoft.graph.eventMessageRequest" || // creation
+		otype == "#microsoft.graph.eventMessage" || // cancellation
+		otype == "#microsoft.graph.eventMessageResponse" { // accept
```

Comment:
the deserialization code is actually pretty generic and relies on graph API to properly interpret things. Could we actually use a type-switch here instead of manually checking the odata type? For example

```go
switch data.(type) {
case models.EventMessageRequestable, models.EventMessageResponseable, models.EventMessageable:
  cal, err := getICalData(ctx, data)
  ...
}
```

Since the switch (may) allow listing multiple types on the same line it could be more compact than multiple if-statements. If it doesn't then I don't have too much of a preference though the switch block may read a little cleaner? You'd need the 2-return value type assertion(?) in golang though (i.e. `msg, ok := data.(models.EventMessageable)`

I guess I'm asking cause I would slightly prefer getting a compile error if the types change rather than just silently skipping the if-block if for some reason the odata types we're checking for are no longer used (yay me being paranoid about things heh)

edit: after looking at the graph SDK more, maybe this can just be an interface type check like
```go
if _, ok := data.(models.EventMessageable) {
  cal, err := getICalData(ctx, data)
  ...
}
```

The different EventMessage types all seem to embed that one type and if we have tests that check the code works for all the types then we'd know if the SDK got updated in some way that broke our expectations